### PR TITLE
PM partitioning for the Direct Xip mode

### PR DESCRIPTION
**Pull request by kapi-no**
Introduced partition description in order to support MCUboot's direct-xip mode.

### CODE REVIEW

File: boot/zephyr/pm.yml [hunk 1]
```diff
@@ -28,15 +28,36 @@ mcuboot_secondary:
...
+  share_size: mcuboot_primary_app
+  placement:
+    after: mcuboot_secondary_pad
+
+mcuboot_secondary:
+  span: [mcuboot_secondary_pad, mcuboot_secondary_app]
+
+#endif /* !defined(CONFIG_SINGLE_APPLICATION_SLOT) && !defined (CONFIG_BOOT_DIRECT_XIP) */
+
+#if CONFIG_BOOT_USING_SCRATCH
```
Comment (nvlsianpu) [inline boot/zephyr/pm.yml:54]:
```suggestion
#if CONFIG_BOOT_SWAP_USING_SCRATCH
```

**Review by nvlsianpu — CHANGES_REQUESTED** [review]

File: boot/zephyr/pm.yml (hunk 1 above)
Comment (kapi-no) [reply boot/zephyr/pm.yml]: fixed

File: boot/zephyr/pm.yml [hunk 2]
```diff
@@ -28,15 +28,36 @@ mcuboot_secondary:
     align_next: CONFIG_FPROTECT_BLOCK_SIZE # Ensure that the next partition does not interfere with this image
     after: mcuboot_primary
 #endif /* CONFIG_PM_EXTERNAL_FLASH_MCUBOOT_SECONDARY */
-#endif /* !defined(CONFIG_SINGLE_APPLICATION_SLOT) */
 
-#if !defined(CONFIG_BOOT_SWAP_USING_MOVE) && !defined(CONFIG_SINGLE_APPLICATION_SLOT) && !defined(CONFIG_BOOT_UPGRADE_ONLY)
+#else
```
Comment (kapi-no) [inline boot/zephyr/pm.yml:32]:
```suggestion
#endif

#if defined(CONFIG_BOOT_DIRECT_XIP)
```

**Comment by mbolivar-nordic** [comment]
Please rebase onto the new history in main

**Comment by kapi-no** [comment]
Rebased after upmerge. The diff is the same

**Review by nvlsianpu — APPROVED** [review]

**Review by de-nordic — APPROVED** [review]

**Pull request merged**